.github/workflows/tests.yml

```diff
@@ -19,7 +19,7 @@ jobs:
   build:
     if: "!(github.event_name == 'push' && github.repository == 'discourse/discourse-private-mirror')"
     name: ${{ matrix.target }} ${{ matrix.build_type }} ${{ matrix.ruby }}
-    runs-on: ${{ (matrix.build_type == 'annotations') && 'ubuntu-latest' || 'ubuntu-20.04-8core' }}
+    runs-on: ${{ (matrix.build_type == 'annotations') && 'ubuntu-latest' || (matrix.target == 'plugins') && (matrix.build_type == 'system') && 'buildjet-16vcpu-ubuntu-2204' || 'buildjet-8vcpu-ubuntu-2204' }}
```

Comment:
I think we can improve the readability here by adding brackets to clarify the order of operations.

Reply:
Brackets per runner type?

Reply:
See if this works

Reply:
FWIW I did a runner matrix as well but the need to exclude so many types of builds was a little uglier.